server/enhancedDate.js

```diff
@@ -1 +1,77 @@
-// export this as a Node module using the enhancedDate.js code from Module_2
\ No newline at end of file
+'use strict';
+
+// Private data, not accessible outside the enhancedDate module
+var days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 
+            'Saturday'];
+var months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 
+            'August', 'September', 'October', 'November', 'December'];
+var date;
+
+// check if date has been set, otherwise create a new date object
+function hasDate() {
+    
+    if (!date) {
+        date = new Date();
+    }
+}
+
+// Methods availabls as a public API    
+// setDate allows users to change the date without touching our hidden date variable
+var setDate = function setDate(dateValue) {
+
+    hasDate();
+    if (typeof dateValue === 'number' || dateValue instanceof Date) {
+        date = new Date(dateValue);
+    }
```

Comment:
setDate should, at a minimum, instantiate a new Date with no arguments if user provides no arguments